lib/backend/k8s/conversion.go

```diff
@@ -209,21 +223,21 @@ func (c converter) podToWorkloadEndpoint(pod *kapiv1.Pod) (*model.KVPair, error)
 	workload := fmt.Sprintf("%s.%s", pod.ObjectMeta.Namespace, pod.ObjectMeta.Name)
 
 	// If the pod doesn't have an IP address yet, then it hasn't gone through CNI.
-	ipNets := []cnet.IPNet{}
-	if c.hasIPAddress(pod) {
-		// Parse the Pod's IP address.
-		_, ipNet, err := cnet.ParseCIDR(fmt.Sprintf("%s/32", pod.Status.PodIP))
-		if err != nil {
-			return nil, err
-		}
-		ipNets = []cnet.IPNet{*ipNet}
+	if !c.hasIPAddress(pod) {
```

Comment:
Should we just be calling our to isCalicoPod() here?

Also, it feels like this will cause errors to trickle up in the list processing and syncer?  Is that not the case?    e.g. in the syncer there is a call to pofdToWorkloadEndpoint() that ignores the errors - is there a chance it will get called for non calico pods and thus error resulting in a panic?